mono_repo/lib/src/commands/travis.dart

```diff
@@ -101,88 +204,78 @@ Future generateTravisConfig(
     'echo -e "${red.wrap("Not expecting TASK '\${TASK}'. Error!")}"',
     'exit 1'
   ]);
+  return taskEntries;
+}
 
-  var envEntries = environmentVars.keys.toList()..sort();
+List<String> _calculatePkgEntries(Map<String, TravisConfig> configs) {
+  var pkgEntries = <String>[];
 
-  var matrix = [];
+  void addEntry(String label, List<String> contentLines) {
+    assert(contentLines.isNotEmpty);
+    contentLines.add(';;');
 
-  /// Iterate in the already sorted order instead of using `forEach`.
-  for (var envVarEntry in envEntries) {
-    var entrySdks = environmentVars[envVarEntry];
-    var excludeSdks = sdks.toSet()..removeAll(entrySdks);
+    var buffer = new StringBuffer('$label) ${contentLines.first}\n');
+    buffer.writeAll(contentLines.skip(1).map((l) => '  $l'), '\n');
 
-    if (excludeSdks.isNotEmpty) {
-      if (matrix.isEmpty) {
-        matrix.addAll(['', 'matrix:', '  exclude:']);
-      }
+    var output = buffer.toString();
+    if (!pkgEntries.contains(output)) {
+      pkgEntries.add(output);
+    }
+  }
 
-      for (var sdk in excludeSdks) {
-        matrix.add('    - dart: $sdk');
-        matrix.add('      env: $envVarEntry');
-      }
+  for (var pkg in configs.keys) {
+    var config = configs[pkg];
+    if (config.beforeScript != null) {
+      addEntry(pkg, [
+        'echo',
+        'echo -e "${styleBold.wrap("PKG: $pkg")}"',
+        config.beforeScript
+      ]);
     }
   }
 
-  var firstAllow = true;
-  var allowFailuresEntries = allowFailures.keys.toList()..sort();
-  for (var envVarEntry in allowFailuresEntries) {
-    var failureSdks = allowFailures[envVarEntry];
+  addEntry('*', [
+    'echo -e "No before_script specified for PKG \'\${PKG}\'."',
```

Comment:
I find this a bit annoying. Better to log when a before script is called – instead of adding noise in the common case.

Reply:
https://github.com/dart-lang/mono_repo/issues/39